### DISCUSSION TITLE
V8: onResizeEnd event

### DISCUSSION THREAD

**Discussion by fknop** [TanStack/table#4235] [Ideas] [answered]
Hello,

I have a use case which I think the library doesn't support (yet?).
I would like to store the column sizing values in local storage at the end of the resizing event while still having the `onChange` behavior. I want to load values from localStorage and pass it to the `initialState`. I do not want to have the value controlled (for performance reason). However, I do not have access to (only) the resizing end event.

Would there be a way to hook into the end of the resizing event while in `onChange` mode so I can store the new values in local storage ? Passing `onColumnSizingChange` makes the `columnSizing` controlled automatically.

Making the component controlled cause a little bit of UI lag compared to the uncontrolled version, which I would like to avoid.

Maybe the solution would be to add a `onColumnSizingEnd` ?

The code I think of would look something like this:

```typescript
const [columnSizing, setColumnSizing] =
  useLocalStorageState<ColumnSizingState>(
    `local-storage-id`,
    {
      defaultValue: {},
    }
  )

const table = useReactTable({
    data,
    columns,
    initialState: {
      columnSizing,
    },
    columnResizeMode: 'onChange',
    onColumnSizingEnd: setColumnSizing,
    getCoreRowModel: getCoreRowModel(),
   // ...
  })

```

**Comment by fknop** [comment, marked as answer]
Well, I kinda feel dumb as I didn't think about that earlier,
But I simply ended up adding this on the resizer element :) 

```typescript
onMouseUp={() => {
  setColumnSizing(table.getState().columnSizing)
}}
```

**Reply by vaziliybober** [reply]
This event isn't fired when you release the mouse button outside of the actual resizer, unfortunately

**Reply by vaziliybober** [reply]
However this seems to work ok

```ts
onMouseDown={(e) => {
  document.addEventListener(
    "mouseup",
    () => console.log(header.getSize()),
    { once: true }
  );
  header.getResizeHandler()(e);
}}
```

**Reply by fknop** [reply]
You're right, I just had the issue :)

**Comment by tannerlinsley** [comment]
You could also watch the table state for changes and trigger a side effect.
